deployment/7_upgrade_ens.js

```diff
@@ -0,0 +1,46 @@
+const ENSManager = require('../build/ENSManager');
+const WalletFactory = require('../build/WalletFactory');
+
+const deploy = async (network) => {
+    const manager = new DeployManager(network);
+    await manager.setup();
+    const configurator = manager.configurator;
+    const deployer = manager.deployer;
+    const abiUploader = manager.abiUploader;
+
+    const config = configurator.config;
+    const domain = config.ENS.domain;
+
+    // Instantiate the ENS Registry and existing WalletFactory and ENSManager
+    const WalletFactoryWrapper = await deployer.wrapDeployedContract(WalletFactory, config.contracts.WalletFactory);
+    const ENSManagerWrapper = await deployer.wrapDeployedContract(ENSManager, config.contracts.ENSManager);
+
+    // Deploy the updated ENSManager
+    const ENSManagerNew = await deployer.deploy(ENSManager, {}, domain, utils.namehash(domain), config.ENS.ensRegistry, config.contracts.ENSResolver);
+
+    const ENSManagerAddManagerTx = await ENSManagerNew.contract.addManager(config.contracts.WalletFactory);
+    await ENSManagerNew.verboseWaitForTransaction(ENSManagerAddManagerTx, 'Set the WalletFactory as the manager of the ENS Manager');
+
+    const WalletFactoryChangeENSManagerTx = await WalletFactoryWrapper.contract.changeENSManager(ENSManagerNew.contractAddress);
+    await WalletFactoryWrapper.verboseWaitForTransaction(WalletFactoryChangeENSManagerTx, `Change the ENSManager on the WalletFactory`);
+
+    const ChangeENSManagerOwnerTx = await ENSManagerNew.contract.changeOwner(config.contracts.MultiSigWallet);
+    await ENSManagerNew.verboseWaitForTransaction(ChangeENSManagerOwnerTx, `Set the MultiSig as the owner of ENSManager}`);
+
+    const ChangeDomainOwnerTx = await ENSManagerWrapper.contract.changeRootnodeOwner("0x0000000000000000000000000000000000000000");
```

Comment:
Is this what we want? If we throw away ownership in the old registry there will be an interruption of service between the moment we execute that transaction and the moment we claim ownership in the new registry. During that interruption of service we won't be able to create wallets and register subdomains... which means we need to coordinate with the backend team (so they disable the creation of wallets).

On the other hand if we properly transfer ownership to the new Manager in the old registry there is no interruption of service (we can always create wallets and register subdomains). The old registry will just be ignored once the new one is setup.